components/style/values/specified/position.rs

```diff
@@ -49,25 +49,46 @@ define_css_keyword_enum! { Y:
 }
 add_impls_for_keyword_enum!(Y);
 
+/// Modern position syntax supports 3 and 4-value syntax. That means:
+/// If three or four values are given, then each <percentage> or <length> represents an offset
+/// and must be preceded by a keyword, which specifies from which edge the offset is given.
+/// For example, `bottom 10px right 20px` represents a `10px` vertical
+/// offset up from the bottom edge and a `20px` horizontal offset leftward from the right edge.
+/// If three values are given, the missing offset is assumed to be zero.
+/// But for some historical reasons we need to keep CSS Level 2 syntax which only supports up to
+/// 2-value. This enum represents whether position should parse modern syntax or legacy syntax.
+#[cfg_attr(feature = "servo", derive(HeapSizeOf))]
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum PositionSyntax {
+    /// Modern syntax
+    Modern,
+    /// Legacy syntax
+    Legacy,
+}
```

Comment:
This is not a big code duplication, this is making sure we can't have values that aren't even representable on the CSS front. Cf. for example, `style::values::specified::transform::OriginComponent`.